app/controllers/audits_controller.rb

```diff
@@ -2,14 +2,18 @@
 
 class AuditsController < ApplicationController
   def index
-      @history = AuditHistory.new(audit_log)
+    @history = AuditHistory.new(audit_log)
   end
 
   private
 
   def audit_log
     if params.key?(:entry_id)
-      return Entry.find(params[:entry_id]).audits
+      if params[:entry_type] == "hours"
+        return Hour.find(params[:entry_id]).audits
+      elsif params[:entry_type] == "mileages"
+        return Mileage.find(params[:entry_id]).audits
+      end
```

Comment:
Can't we use constantize for this?
http://apidock.com/rails/Inflector/constantize